### PR TITLE
ci: Modifies release action

### LINKED ISSUE #651: Add versioning to project
# Motivation

We should add versioning to this project along with a changelog to enable users to easily see what's changed between deployments.

This will also allow us to make breaking changes by doing major version bumps, along with adding migration guides in the changelog.

# How would you feel if this feature request was implemented?

changes

# Requirements

- Automated versioning and GitHub release when merges to main
- Documentation on how to do major version changes
- Automated changelog generation

# Tasks

- [ ] #805 
- [x] #829 
- [x] #832
- [x] #837
- [x] #838
- [x] #859
- [ ] #907
- [ ] #909
- [ ] #910

### PR DESCRIPTION
**Pull request by frtibble**
## Purpose
This PR uses the semantic release action to create a release. It does so by:
- Analysing commit messages
- Determining version number (if a release is required)
- It *does not* update the changelog (this would require permissions for the GitHub action to push to `main` which we cannot enable due to branch protections)
- It uses the version number to push a tag, which creates a release

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

Resolves #651

### CODE REVIEW

**Review by adamdougal — DISMISSED** [review]
lgtm

File: .github/workflows/create-release.yml [hunk 1]
```diff
@@ -3,6 +3,7 @@ on:
     workflows: ["CI"]
     types:
       - completed
+  merge_group:
```
Comment (cecheta) [inline .github/workflows/create-release.yml:6]: Is this necessary? Doesn't this mean it essentially runs on PRs?

Comment (frtibble) [reply .github/workflows/create-release.yml]: removed!

**Review by cecheta — APPROVED** [review]
🤞

**Pull request merged**